src/Cli/dotnet/commands/dotnet-info/InfoCommand.cs

```diff
@@ -0,0 +1,197 @@
+            WorkloadCommandParser.ShowWorkloadsInfo();
+        }
+
+        private static string GetDisplayRid(DotnetVersionFile versionFile)
+        {
+            FrameworkDependencyFile fxDepsFile = new();
+
+            string currentRid = RuntimeInformation.RuntimeIdentifier;
+
+            // if the current RID isn't supported by the shared framework, display the RID the CLI was
+            // built with instead, so the user knows which RID they should put in their "runtimes" section.
+            return fxDepsFile.IsRuntimeSupported(currentRid) ?
+                currentRid :
+                versionFile.BuildRid;
+        }
+
+        public class RuntimeEnvironmentInfo
+        {
+            public string Name { get; set; }
+            public string Version { get; set; }
+            public string Platform { get; set; }
+            public string Rid { get; set; }
+        }
+
+        public class SdkInfo
+        {
+            public string Commit { get; set; }
+            public string Version { get; set; }
+        }
+        public class HostInfo
+        {
+            public string Arch { get; set; }
+            public string Commit { get; set; }
+            public string Version { get; set; }
+        }
+
+        public class InstalledRuntime
+        {
+            public string Name { get; set; }
+            public string Path { get; set; }
+            public string Version { get; set; }
+        }
+
+        public class OtherArchInfo
+        {
+            public string Arch { get; set; }
+            public string BasePath { get; set; }
+        }
+
+        public class WorkloadManifest
+        {
+            public string InstallType { get; set; }
+            public string Path { get; set; }
+            public string Version { get; set; }
+        }
+
+        public class Workload
+        {
+            public List<InstallSource> InstallSources { get; set; }
+            public WorkloadManifest Manifest { get; set; }
+            public string Name { get; set; }
+        }
+
+        public class InstallSource
+        {
+            public string Name { get; set; }
+            public string Version { get; set; }
+        }
+
+        public class DotNetInfo
+        {
+            public string BasePath { get; set; }
+            public Dictionary<string, string> EnvVars { get; set; }
+            public string GlobalJson { get; set; }
+            public HostInfo Host { get; set; }
+            public List<InstalledRuntime> InstalledRuntimes { get; set; }
+            public List<OtherArchInfo> OtherArch { get; set; }
+            public RuntimeEnvironmentInfo RuntimeEnv { get; set; }
+            public SdkInfo Sdk { get; set; }
+            public List<Workload> Workloads { get; set; }
+
+            public string ToJson()
+            {
+                return JsonSerializer.Serialize(this, new JsonSerializerOptions { WriteIndented = true });
```

Comment:
For this relatively simple JSON content personally I would go with just `Utf8JsonWriter`. Maintaining the classes just for serialization feels like an overkill (and it will need lot of attributes to change the property names to camel case and so on).

There's also a performance consideration. Currently this uses reflection based serialization which brings in a lot of dependencies on the reflection stack which is pretty expensive. Since we're writing this as new code, we should try to make it better than that. If you decide to still go with object based serialization, then this should use the source generator. If you go with the direct JSON writer approach the problem goes away as well.